src/business_logic/state/cached_state.rs

```diff
@@ -152,6 +172,56 @@ impl<T: StateReader + Clone> StateReader for CachedState<T> {
         let modified_contracts = storage_updates.keys().map(|k| k.0.clone()).len();
         (modified_contracts, storage_updates.len())
     }
+
+    fn get_compiled_class(
+        &mut self,
+        compiled_class_hash: &ClassHash,
+    ) -> Result<CompiledClass, StateError> {
+        if let Some(casm_class) = &mut self.casm_contract_classes {
+            if let Some(class) = casm_class.get(compiled_class_hash) {
+                return Ok(CompiledClass::Casm(Box::new(class.clone())));
+            }
+        }
+        if let Some(contract_class) = &mut self.contract_classes {
+            if let Some(class) = contract_class.get(compiled_class_hash) {
+                return Ok(CompiledClass::Deprecated(Box::new(class.clone())));
+            }
+        }
```

Comment:
It would seem from here that it can only be one of the two. Wouldn't it make sense to use an `enum` for that? That way you'd have a single `match` here.